### PR TITLE
Clean up and re-organize the docs folder

### PR DESCRIPTION
**Pull request by dnadales**
This PR contains multiple commits that can be reviewed in isolation. I can create stacked PRs if needed.

Closes https://github.com/input-output-hk/ouroboros-network/issues/4217
Closes https://github.com/input-output-hk/ouroboros-network/issues/4061

### CODE REVIEW

File: docs/References.md [hunk 1]
```diff
@@ -0,0 +1,28 @@
+# References 
+
+The following artifacts influence and/or describe the Consensus implementation.
+
+* [The Cardano Consensus and Storage Layer][consensus-report].
+* [Introduction to the design of Data Diffusion and Networking of Cardano
+  Shelley][network-report].
+* From the IOG researchers:
+    * [Ouroboros BFT][ouroboros-bft] 
+    * [Ouroboros Pros][ouroboros-praos]
```
Comment (Jasagredo) [inline docs/References.md:10]:
```suggestion
    * [Ouroboros Praos][ouroboros-praos] 
```

File: docs/initialization-steps-and-the-related-tracer.org [hunk 2]
```diff
@@ -5,6 +5,9 @@
 #+LATEX_HEADER: \usepackage[margin=0.9in]{geometry}
 #+OPTIONS: toc:nil
 
+FIXME: this contains way too many code details and it is likely to bitrot. Maybe
```
Comment (Jasagredo) [inline docs/initialization-steps-and-the-related-tracer.org:8]: I would move whatever we can extract from this file into the source code and then delete it.

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -0,0 +1,178 @@
...
+sudo mkdir -p /etc/nix
+cat <<EOF | sudo tee -a /etc/nix/nix.conf
+substituters = https://cache.nixos.org https://cache.iog.io
+trusted-public-keys = hydra.iohk.io:f/Ea+s+dFdN+3Y/G+FDgSq+a5NEWhJGzdjvKNGv0/EQ= cache.nixos.org-1:6NCHdD59X431o0gWypbMrAURkbJ16ZPMQFGspcDShjY=
+EOF
+```
+
+## Using `cabal`
+
+A more involved alternative to using `nix` is to set up the developmen
```
[the remark this answers is not in the document]
Comment (Jasagredo) [reply CONTRIBUTING.md:60]: I wouldn't personally say this is more involved TBH.

**Comment by dnadales** [comment]
Relevant PR https://github.com/input-output-hk/ouroboros-network/pull/4249/files

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -0,0 +1,178 @@
...
+```
+
+As an optional step, to improve build speed, you can set up a binary
+cache maintained by IOG:
+
+```sh
+sudo mkdir -p /etc/nix
+cat <<EOF | sudo tee -a /etc/nix/nix.conf
+substituters = https://cache.nixos.org https://cache.iog.io
+trusted-public-keys = hydra.iohk.io:f/Ea+s+dFdN+3Y/G+FDgSq+a5NEWhJGzdjvKNGv0/EQ= cache.nixos.org-1:6NCHdD59X431o0gWypbMrAURkbJ16ZPMQFGspcDShjY=
```
[the remark this answers is not in the document]
Comment (dnadales) [reply CONTRIBUTING.md:54]: But not this trusted public key as far as I recall. I took this from the latest `cardano-node` [build instructions](https://github.com/input-output-hk/cardano-node/blob/master/doc/getting-started/building-the-node-using-nix.md).

File: CONTRIBUTING.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (dnadales) [reply CONTRIBUTING.md:60]: Ups, I hit a nerve :stuck_out_tongue: But good point. I'll remove the adjective.

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -0,0 +1,178 @@
...
+in the command line, either inside `nix-shell` if you use `nix`, or in a
+system with `cabal` installed.
+
+# Setting up your developer environment
+
+TODO: we do not support a single editor, however can we give some
+generic instructions here? Alternatively we could propose **a** way to
+setup each of the editors we use (Emacs, VSCode, Vim), just to give the
+reader something to hold on to if they’re clueless. Otherwise we could
+skip this version.
```
[the remark this answers is not in the document]
Comment (dnadales) [reply CONTRIBUTING.md:82]: Alright, I skipped it.

File: CONTRIBUTING.md [hunk 6]
```diff
@@ -0,0 +1,178 @@
...
+will guide you through the process when reviewing your pull request.
+
+## Following our git process
+
+Our [git process](docs/GitProcess.xml) describes the `git` practices we
+encourage when working with the code in this repository.
+
+## Following the style guide
+
+We have a [Haskell style guide](STYLEGUIDE.xml) that should be followed
```
[the remark this answers is not in the document]
Comment (dnadales) [reply CONTRIBUTING.md:109]: Ups, this is a survivor of a previous iteration.

File: CONTRIBUTING.md [hunk 7]
```diff
@@ -0,0 +1,178 @@
...
+when writing code in Consensus. Our style guide is not set in stone, and
+improvements are always welcome.
+
+## Formatting the code
+
+We use `stylish-haskell` for Haskell code formatting. There is a CI
```
[the remark this answers is not in the document]
Comment (dnadales) [reply CONTRIBUTING.md:115]: That'd be useful indeed. Should we add this as a separate issue?

[the remark this answers is not in the document]
Comment (Jasagredo) [reply CONTRIBUTING.md:115]: Done in a separate issue

File: docs/AnOverviewOfConsensus.md [hunk 8]
```diff
@@ -0,0 +1,64 @@
...
+    transactions as possible.
+
+  * The net should be very difficult to disrupt. For example, a successful
+    denial of service attack should be prohibitively expensive to enact.
+
+The primary data are _blocks_ and _transactions_. The possible contents of both
+of those are primarily constrained by the ledger rules. The Consensus Layer and
+[Network Layer][ouroboros-network] together
+implement a _filtering forwarding network_. Specifically, the Consensus Layer
+determines which blocks propagate between neighboring nodes: those on the _best_
```
[the remark this answers is not in the document]
Comment (dnadales) [reply docs/AnOverviewOfConsensus.md:25]: Hmm, I thought this was still true. @amesgen, could you clarify this bit?

File: docs/AnOverviewOfConsensus.md [hunk 9]
```diff
@@ -0,0 +1,64 @@
...
+chain. The IOG researchers have established that the Ouroboros protocol can be
+used to ensure that -- unless an adversary controls more than half of the net's
+stake -- the honest nodes will all continually reach _consensus_ regarding the
+selection of a single best chain and that that chain grows over time.
+
+The Consensus Layer defines the core Consensus components and logic, notably the
+Ouroboros protocol.
```
[the remark this answers is not in the document]
Comment (dnadales) [reply docs/AnOverviewOfConsensus.md:32]: I added a link to `docs/References.md`. Is that ok?

[the remark this answers is not in the document]
Comment (Jasagredo) [reply docs/AnOverviewOfConsensus.md:32]: Yup

File: docs/AnOverviewOfConsensus.md (hunk 8 above)
[the remark this answers is not in the document]
Comment (amesgen) [reply docs/AnOverviewOfConsensus.md:25]: Yeah, the _best_ chain sounds like something that is guaranteed to be valid, but what we announce via `ChainSync` is not necessarily valid when we diffuse a "trap" tentative header. Still, I think this is fine for a "very high-level motivation"; it does not seem unreasonable (or even inevitable) to use slightly fuzzy language in such a context.

File: CONTRIBUTING.md (hunk 7 above)
[the remark this answers is not in the document]
Comment (dnadales) [reply CONTRIBUTING.md:107]: https://github.com/input-output-hk/ouroboros-consensus/issues/2

File: docs/AnOverviewOfConsensus.md (hunk 8 above)
[the remark this answers is not in the document]
Comment (dnadales) [reply docs/AnOverviewOfConsensus.md:25]: Thank you guys!

**Review by Jasagredo — APPROVED** [review]

**Pull request merged**